From 6b3cfcd1826cc6b821061ca5cd15e405cab26fd0 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Wed, 27 Aug 2025 10:31:29 +0800
Subject: [PATCH 39/63] loonggpu: Get rid of drm_sched_job.id

It's removed in the upstream kernel.  I don't bother to add a #ifdef for
kernel version check here because our users don't care about tracing
anyway, and developers should always use the latest kernel.

Fix build failure on AOSC 6.17-rc3 kernel.

Link: https://git.kernel.org/torvalds/c/4f7fa5fa414c
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 include/loonggpu_trace.h | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/include/loonggpu_trace.h b/include/loonggpu_trace.h
index 0e476e9..8d85ae2 100644
--- a/include/loonggpu_trace.h
+++ b/include/loonggpu_trace.h
@@ -142,7 +142,6 @@ TRACE_EVENT(loonggpu_cs_ioctl,
 	    TP_PROTO(struct loonggpu_job *job),
 	    TP_ARGS(job),
 	    TP_STRUCT__entry(
-			     __field(uint64_t, sched_job_id)
 			     __field(unsigned int, context)
 			     __field(unsigned int, seqno)
 			     __field(struct dma_fence *, fence)
@@ -151,14 +150,13 @@ TRACE_EVENT(loonggpu_cs_ioctl,
 			     ),
 
 	    TP_fast_assign(
-			   __entry->sched_job_id = job->base.id;
 			   __entry->context = job->base.s_fence->finished.context;
 			   __entry->seqno = job->base.s_fence->finished.seqno;
 			   __entry->ring_name = to_loonggpu_ring(job->base.sched)->name;
 			   __entry->num_ibs = job->num_ibs;
 			   ),
-	    TP_printk("sched_job=%llu, context=%u, seqno=%u, ring_name=%s, num_ibs=%u",
-		      __entry->sched_job_id, __entry->context,
+	    TP_printk("context=%u, seqno=%u, ring_name=%s, num_ibs=%u",
+		      __entry->context,
 		      __entry->seqno, __entry->ring_name, __entry->num_ibs)
 );
 
@@ -166,7 +164,6 @@ TRACE_EVENT(loonggpu_sched_run_job,
 	    TP_PROTO(struct loonggpu_job *job),
 	    TP_ARGS(job),
 	    TP_STRUCT__entry(
-			     __field(uint64_t, sched_job_id)
 			     __field(unsigned int, context)
 			     __field(unsigned int, seqno)
 			     __field(char *, ring_name)
@@ -174,14 +171,13 @@ TRACE_EVENT(loonggpu_sched_run_job,
 			     ),
 
 	    TP_fast_assign(
-			   __entry->sched_job_id = job->base.id;
 			   __entry->context = job->base.s_fence->finished.context;
 			   __entry->seqno = job->base.s_fence->finished.seqno;
 			   __entry->ring_name = to_loonggpu_ring(job->base.sched)->name;
 			   __entry->num_ibs = job->num_ibs;
 			   ),
-	    TP_printk("sched_job=%llu, context=%u, seqno=%u, ring_name=%s, num_ibs=%u",
-		      __entry->sched_job_id, __entry->context,
+	    TP_printk("context=%u, seqno=%u, ring_name=%s, num_ibs=%u",
+		      __entry->context,
 		      __entry->seqno, __entry->ring_name, __entry->num_ibs)
 );
 
-- 
2.52.0

